pkg/policyfilter/state.go

```diff
@@ -105,28 +110,33 @@ func (pod *podInfo) delContainers(id string) []containerInfo {
 
 // containerExists checks returns true if a container exists in the pod
 // it uses log to log inconsistencies in the data.
-func (pod *podInfo) containerExists(log logrus.FieldLogger, containerID string, cgID CgroupID) bool {
+func (pod *podInfo) containerExists(log logrus.FieldLogger, containerID string, cgIDp *CgroupID) bool {
 	for i := range pod.containers {
 		container := &pod.containers[i]
-		if container.id == containerID && container.cgID == cgID {
-			// container was already handled, return
+		if container.id != containerID {
+			continue
+		}
+
+		// we found a matching container id, if no cgroup id is given, return
+		if cgIDp == nil {
```

Comment:
Just to understand under which conditions this could happen not having a cgIDp?

Reply:
It's explained in the commit message that adds the pod event handlers:

> 
>     pkg/policyfilter: add pod event handlers
>     
>     This patch adds pod event handlers for policyfilter.
>     
>     A new function RegisterPodHandlers is exported, which allows callers to
>     register the needed hooks.
>     
>     The hooks monitor pod additions and deletions, and update the
>     policyfilter accordingly. These handlers may work i) in combination with
>     run-time hooks, or ii) in the absence of run-time hooks. Note in case of
>     ii) there is a race.
>     
>     Removing containers is easy, we just remove the state for a given
>     container id.
>     
>     Adding containers is more involved because we do not know the cgroup id
>     to add to the policyfilter bpf map from the k8s API information. If a run-time hook was already
>     executed, our state should be up-to-date, so there is no more work to do.
>     However, there are cases where this might not have happened. One example
>     of this is when the agent starts, and the existing pods are processed.
>     Another example is case ii) as described above.
>     
>     For these cases, we need to find the cgroup id from the container name.
>     To do so, we use cgroups.fsscan. Hence, the function AddPodContainer is
>     modified to take a pointer to a cgroup id. If that pointer is not-nil,
>     everything works as before. Otherwise, the code will try to determine

Hope this explains why we need it.